app/views/common/os_selection/_architecture.html.erb

```diff
@@ -3,7 +3,8 @@
     {:selected => item.operatingsystem_id, :include_blank => blank_or_inherit_f(f, :operatingsystem) },
     { :label => _("Operating system"),
       :help_inline => :indicator,
-      :onchange => 'os_selected(this);', :'data-url' => method_path('os_selected'), :'data-type' => controller_name.singularize, :required => true,
+      :onchange => 'os_selected(this);', :'data-url' => method_path('os_selected'), :'data-type' => controller_name.singularize,
+      :required => f.object.is_a?(Hostgroup) ? false :   true,
```

Comment:
There are some extra spacing here and in the other file between `:` and `true`. 
Also, note that `f.object.is_a?(Hostgroup)` returns a boolean, so instead of using a trenary operator, you can just negate the result. 
Finally, I think you can use `item` instead of `f.object`.